<a href="https://colab.research.google.com/github/purnawanpp/plants/blob/main/plants.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Cara Menjalankan Software Protein-Ligand ANTSystem (PLANTS) di google colab by Purnawan Pontana Putra**

**Instalasi Software yang dibutuhkan**

In [1]:
!wget https://github.com/purnawanpp/plants/raw/main/plants
!wget https://github.com/purnawanpp/plants/raw/main/spores
!apt install openbabel
!chmod +x /content/plants
!chmod +x /content/spores
!wget https://github.com/purnawanpp/plants/raw/main/dock_gcolab.txt

--2023-03-08 06:15:38--  https://github.com/purnawanpp/plants/raw/main/plants
Resolving github.com (github.com)... 20.27.177.113
Connecting to github.com (github.com)|20.27.177.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/purnawanpp/plants/main/plants [following]
--2023-03-08 06:15:39--  https://raw.githubusercontent.com/purnawanpp/plants/main/plants
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.111.133, 185.199.110.133, 185.199.108.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.111.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 9536847 (9.1M) [application/octet-stream]
Saving to: ‘plants’

plants              100%[===================>]   9.09M  --.-KB/s    in 0.1s    

2023-03-08 06:15:41 (84.3 MB/s) - ‘plants’ saved [9536847/9536847]

--2023-03-08 06:15:41--  https://github.com/purnawanpp/plants/raw/main/spores
Resol

**Download Protein PDB ID 3HTB**

In [2]:
!wget https://files.rcsb.org/download/3HTB.pdb

--2023-03-08 06:16:07--  https://files.rcsb.org/download/3HTB.pdb
Resolving files.rcsb.org (files.rcsb.org)... 128.6.158.49
Connecting to files.rcsb.org (files.rcsb.org)|128.6.158.49|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/octet-stream]
Saving to: ‘3HTB.pdb’

3HTB.pdb                [  <=>               ] 161.13K   302KB/s    in 0.5s    

2023-03-08 06:16:08 (302 KB/s) - ‘3HTB.pdb’ saved [164997]



**Separete Protein and ligand**

In [3]:
!./spores --mode splitpdb 3HTB.pdb


				 SPORES        

	      Structure PrOtonation and REcognition System

			      version 1.3


author: T. ten Brink

scientific contributors: P. Duchstein, J. Gimmler, O. Korb, M. Zerara,
			  T.E. Exner

contact: tim.tenbrink@uni-konstanz.de


molecule number: 1
64 alternative atomic positions have been removed !
splitting 3HTB
225
protein: 0
ligs: 4
water: 660
metal: 0


**Add Hydrogen, charge (gastaiger) dan protonasi pada pH 7.4**

Keterangan:

1. -p 7.4: menetapkan pH 7.4 untuk menghitung muatan.
2. -b: menghitung muatan terbentuk pada pH yang ditetapkan.
3. --ff "gasteiger": menetapkan metode perhitungan muatan menggunakan metode Gasteiger.

In [4]:
!obabel -imol2 /content/protein.mol2 -omol2 -h -p 7.4 -b --ff "gasteiger" -O rec.mol2
!obabel -imol2 /content/ligand_JZ4167_0.mol2 -omol2 -h -p 7.4 -b --ff "gasteiger" -O lig.mol2

*** Open Babel Warning  in ReadMolecule
  Failed to kekulize aromatic bonds in MOL2 file (title is 3HTB_protein)

*** Open Babel Warning  in DoTransformations
  Both -p and -h options are set. All implicit hydrogens (-h) will be added without considering pH.

1 molecule converted
*** Open Babel Warning  in DoTransformations
  Both -p and -h options are set. All implicit hydrogens (-h) will be added without considering pH.

1 molecule converted


**Membuat Binding Sitenya atau Gridboxnya**

In [5]:
!./plants --mode bind lig.mol2 rec.mol2



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile



Calculated bindingsite definition (from 1 ligands): 

bindingsite_center 22.7142 -25.2627 -3.283
bindingsite_radius 3.75624



Pindahkan data:

*   bindingsite_center 22.7142 -25.2627 -3.283
*   bindingsite_radius 3.75624

kedalam file **dock_gcolab.txt**

**Simulasi Molecular Docking**

In [6]:
!./plants --mode screen /content/dock_gcolab.txt



                  PLANTS
        Protein-Ligand ANT System
               version 1.2


author: Oliver Korb

scientific contributors: T.E. Exner, T. Stuetzle

contact: Oliver.Korb@uni-konstanz.de


run PLANTS: PLANTS --mode screen yourconfigfile

PLANTS warning: Config file parser found unknown token '' in line 2.
PLANTS warning: Please check config file line 2. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token '' in line 6.
PLANTS warning: Please check config file line 6. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token '' in line 10.
PLANTS warning: Please check config file line 10. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token 'flexible_protein_side_chain_11' in line 12.
PLANTS warning: Config file parser found unknown token '' in line 13.
PLANTS warning: Please check config file line 13. Maybe some input is missing.
PLANTS warning: Config file parser found unknown token '' i

Setelah proses selesai, jangan lupa klik simbol kotak/file refresh google colab. Hasil simulasi docking dapat dilihat pada folder *results* dengan nama file **bestranking.csv**, lihat energy TOTAL_SCORE merupakan nilai delta G 

**Perhitungan RMSD hasil simulasi**

In [7]:
!obrms -f lig.mol2 /content/results/3HTB_ligand_2_entry_00001_conf_01.mol2

RMSD 3HTB_ligand_2:3HTB_ligand_2_entry_00001_conf_01 0.505936
